src/plugins/vis_type_vega/public/data_model/types.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { AggParamsMapping, Filter } from 'src/plugins/data/public';
+import { DslQuery } from 'src/plugins/data/common';
+import { EsQueryParser } from './es_query_parser';
+import { EmsFileParser } from './ems_file_parser';
+import { UrlParser } from './url_parser';
+
+interface Body {
+  aggs?: AggParamsMapping;
+  query?: Query;
+  timeout?: string;
+}
+
+interface Coordinate {
+  axis: {
+    title: string;
+  };
+  field: string;
+}
+
+interface Encoding {
+  x: Coordinate;
+  y: Coordinate;
+}
+
+interface AutoSize {
+  type: string;
+  contains: string;
+}
+
+export interface VegaSpec {
+  [index: string]: any;
+  $schema: string;
+  data?: Data;
+  encoding?: Encoding;
+  mark?: string;
+  title?: string;
+  autosize: AutoSize;
+  projections?: any;
+  width?: any;
+  height?: any;
+  padding?: any;
```

Comment:
I couldn't figure out what would be the types of these. I think it would most probably be number. Can I have little help here

Reply:
If you don't know what type they are, just leave them as `unknown`. If they are used in a specific way somewhere in the code, you will get a type error there and can adjust.